source/common/upstream/eds.cc

```diff
@@ -59,20 +60,42 @@ void EdsClusterImpl::onConfigUpdate(const ResourceVector& resources) {
                                      cluster_load_assignment.cluster_name()));
   }
   for (const auto& locality_lb_endpoint : cluster_load_assignment.endpoints()) {
+    uint32_t priority = locality_lb_endpoint.priority().value();
+    if (new_hosts.size() <= priority) {
+      new_hosts.resize(priority + 1);
```

Comment:
Would it make sense to make this structure `std::map`, i.e. ordered, to support sparse priority use? I'm thinking not, since probably folks will use dense priorities in practice, but throwing it out there.

Reply:
Yeah as commented in the eds proto you can't skip.  We don't have config validation so I just make sure the code works regardless but it doesn't make sense to do so reorganizing the data structures to optimize for it seems silly.